cypress/integration/console/devices/onboarding/qr-scan.spec.js

```diff
@@ -0,0 +1,125 @@
+      cy.createApplication(application, user.ids.user_id)
+    })
+
+    beforeEach(() => {
+      cy.fixture('console/devices/repository/test-brand-otaa-model3.template.json').then(
+        templateJson => {
+          cy.intercept(
+            'GET',
+            `/api/v3/dr/applications/${appId}/brands/test-brand-otaa/models/test-model3/1.0.1/EU_863_870/template`,
+            templateJson,
+          )
+        },
+      )
+      cy.fixture('console/devices/repository/test-brand-otaa.models.json').then(modelsJson => {
+        cy.intercept(
+          'GET',
+          `/api/v3/dr/applications/${appId}/brands/test-brand-otaa/models*`,
+          modelsJson,
+        )
+      })
+      cy.fixture('console/devices/repository/brands.json').then(brandsJson => {
+        cy.intercept('GET', `/api/v3/dr/applications/${appId}/brands*`, brandsJson)
+      })
+
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+      cy.visit(`${Cypress.config('consoleRootPath')}/applications/${appId}/devices/add`)
+      cy.findByText('Scan end device code').click()
+    })
+
+    it('succeeds registering a device via a qr code', () => {
+      cy.findByTestId('full-error-view').should('not.exist')
+      cy.findByTestId('error-notification').should('not.exist')
+
+      cy.findByText('Scan end device QR code').should('be.visible')
+      cy.findByText('Please scan the QR code to continue.').should('exist')
+      cy.findByTestId('webcam-feed').should('be.visible')
+
+      cy.findByText('Found QR code data').should('be.visible')
+
+      cy.findByText('Apply').should('not.be.disabled').click()
+
+      cy.findByText('QR code scanned successfully').should('exist')
+      cy.findByText('End device type information').should('exist')
+
+      cy.findByLabelText('End device brand')
+        .parents('.select__control')
+        .next('input')
+        .should('have.value', 'test-brand-otaa')
+
+      selectDevice({
+        model_id: 'test-model3',
+        hw_version: '2.0',
+        fw_version: '1.0.1',
+        band_id: 'EU_863_870',
+      })
+
+      // End device registration.
+      cy.findByLabelText('Frequency plan').selectOption('EU_863_870_TTN')
+      cy.findByLabelText('JoinEUI').should('have.value', device.join_eui).should('be.disabled')
+      cy.findByLabelText('DevEUI').should('have.value', device.dev_eui).should('be.disabled')
+      cy.findByRole('button', { name: 'Reset' }).should('be.disabled')
+      cy.findByLabelText('AppKey').type(generateHexValue(32))
+      cy.findByLabelText('End device ID').should(
+        'have.value',
+        `eui-${device.dev_eui.toLocaleLowerCase().replace(/\s/g, '')}`,
+      )
+
+      cy.findByRole('button', { name: 'Add end device' }).click()
+
+      cy.location('pathname').should(
+        'eq',
+        `${Cypress.config('consoleRootPath')}/applications/${appId}/devices/eui-${device.dev_eui
+          .toLocaleLowerCase()
+          .replace(/\s/g, '')}`,
+      )
+    })
```

Comment:
Please also wait until the device page was loaded and assert the created values in the device overview